src/semver.nim

```diff
@@ -0,0 +1,85 @@
+##
+## Copyright (c) 2023, Crash Override, Inc.
+##
+## This file is part of Chalk
+## (see https://crashoverride.com/docs/chalk)
+##
+import strutils, util
+
+# very simple semver implementation
+# it this does NOT handle full semver spec
+# it only handles basic dot-separated version format
+
+type Version* = ref object
+    parts: seq[int]
```

Comment:
Why isn't parts a three-tuple that has default values if you want to allow short version strings?  It would make the comparison more efficient and more clear too (Many people aren't going to get the 'zip' idiom anyway).

Reply:
changed to tuple